pkg/webhook/server.go

```diff
@@ -181,6 +180,11 @@ func (s *Server) Register(webhooks ...Webhook) error {
 		}
 		s.registry[webhook.GetPath()] = webhooks[i]
 		s.sMux.Handle(webhook.GetPath(), webhook.Handler())
+
+		// Inject dependencies into each webhook
+		if err := s.manager.SetFields(webhooks[i]); err != nil {
```

Comment:
Interesting... Won't manager call `SetField` on webhook server and that should call the inject field for each of the webhooks ? because manager knows when all the deps are resolved ?

Reply:
yeah, this should be receiving an injectfunc, which it should be using to inject things, as opposed to having a reference to the manager.

Reply:
plus, this should be called before trying to call any methods on the webhook